src/ProjectOrigin.WalletSystem.Server/Activities/TransferFullSliceActivity.cs

```diff
@@ -133,4 +138,10 @@ private static TransferredEvent CreateTransferEvent(WalletSlice sourceSlice, IPu
         };
         return transferredEvent;
     }
+
+    private async Task<WalletAttribute[]> GetWalletAttributees(Guid walletId, Guid certificateId, string registryName, IEnumerable<string> hashedAttributeKeys)
+    {
+        var result = await Task.WhenAll(hashedAttributeKeys.Select(key => _unitOfWork.CertificateRepository.GetWalletAttribute(walletId, certificateId, registryName, key)));
```

Comment:
It would also be possible to make a single DB query taking an array of hashed attribute keys

Reply:
Didn't know that, has been rewritten :)